### ISSUE TITLE
Setup-preCICE GitHub action

### ISSUE THREAD

**Issue by erikscheurer** [precice/precice#1544]
The solverdummy test of the bindings (I know of [python](https://github.com/precice/python-bindings/blob/71e7a6e38af3418192b3d8f65006bc8f97a1d4dd/.github/workflows/build-and-test.yml#L134) and [Julia](https://github.com/precice/PreCICE.jl/blob/a2ee7366295b942cafd00286566dcb7c7bc1d154/.github/workflows/build-and-test.yml#L26)) has to set up preCICE in some way. At the moment they are using the latest docker container as an automatic test. This is quite convenient, but in talk with @IshaanDesai we discussed, that the bindings should always test against precice/develop instead of this docker container. 

The [matlab bindings](https://github.com/precice/matlab-bindings/pull/16#issuecomment-1180648214) make use of [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) to set up matlab and install the latest version of preCICE as to avoid repetition of the same setup code.
To make the test setup easier and prettier in the bindings, I am proposing a setup-preCICE-action, that would sit maybe in precice/precice or in a new repository precice/setup-preCICE-action or something, which would then set up a specified preCICE branch by compiling it from source. 
If such an action existed, we would be able to call:
```yaml
uses: precice/setup-preCICE-action@v1
with: 
    branch: develop
```
or something of this sorts to easily compile preCICE instead of compiling in each binding repo individually.
Is this something that would add to precice? Or would it only make things more complex to maintain with more repos?

**Comment by uekerman** [comment]
I am not sure yet whether we need it, but if so, I would argue to put it in a separate repo.
As repo and action name, try to stick to GitHub action conventions: `setup-precice` for the action, `setup-precice-action` for the repo.

**Comment by BenjaminRodenberg** [comment]
Do you only need an up-to-date version of `precice/develop`? Wouldn't it be sufficient to have a nightly build of `precice/precice:develop`? https://github.com/precice/precice/issues/1115

**Comment by IshaanDesai** [comment]
> Do you only need an up-to-date version of `precice/develop`? Wouldn't it be sufficient to have a nightly build of `precice/precice:develop`?

Yes right now this discussion came up because we need an up-to-date version of `precice/develop` as an image from which we can create a container and run the tests in. Right now at several places we are cloning preCICE and installing from source. This instructions from a source installation can be placed somewhere centrally (what @erikscheurer is suggesting) to avoid repetitive instruction blocks in GitHub Actions files.

**Comment by erikscheurer** [comment]
> Do you only need an up-to-date version of `precice/develop`? Wouldn't it be sufficient to have a nightly build of `precice/precice:develop`? #1115

Yes, such a docker container would resolve this, so this issue can be marked as a duplicate. Depending on the computational effort and time complexity, my suggestion could be an easy alternative, but I'm not to closely familiar with the docker procedure.

**Comment by uekerman** [comment]
Could indeed be simpler, more elegant and more recent than the approach with a Docker container.
The "more recent" could indeed be helpful in the v2 --> v3 transition phase.

**Comment by MakisH** [comment]
This addresses almost the same issue as https://github.com/precice/precice/pull/1090 (will revive it soon, together with @valentin-seitz).

In some extent, I see some additional value here: publishing a preCICE GitHub Action could be useful for others. But it should more about a workflow than about a container.

**Comment by uekerman** [comment]
Why not just giving it a try? Could also just be interesting to learn.

**Comment by IshaanDesai** [comment]
@erikscheurer I'll create the repo, lets give this a shot.

**Comment by erikscheurer** [comment]
To give an update: The [repo](precice/setup-precice-action) is up and running, I created a [PR](precice/setup-precice-action#1) there, that is supposed to bring the composite action into a working state. In this pr is also a better description on how the action is set up. At the moment, the compilation works fine including the tests. 

I am also testing the action in the [Julia bindings](precice/PreCICE.jl#197) and I'm getting an error saying that the action is not adding the preCICE installation to the `LD_LIBRARY_PATH`. If someone knows how the environment variables are handled in between composite workflows and the calling workflow, help would be much appreciated.

Should I give more detailed updates in this issue or restrict that to the repo itself?

**Comment by erikscheurer** [comment]
Correction, the installation tests for `precice/main` work fine, the tests for `precice/develop` fail, did something regarding the installation change?

**Comment by IshaanDesai** [comment]
We now have such an action: https://github.com/precice/setup-precice-action
